boot/bootutil/src/loader.c

```diff
@@ -773,6 +773,39 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
         goto out;
     }
 
+#if BOOT_IMAGE_NUMBER > 1 && !defined(MCUBOOT_ENC_IMAGES) && defined(MCUBOOT_VERIFY_IMG_ADDRESS)
+    /* Verify that the image in the secondary slot has a reset address
+     * located in the primary slot. This is done to avoid users incorrectly
+     * overwriting an application written to the incorrect slot.
+     */
+    if (area_id == FLASH_AREA_IMAGE_SECONDARY(BOOT_CURR_IMG(state))) {
+        const struct flash_area *pri_fa = BOOT_IMG_AREA(state, BOOT_PRIMARY_SLOT);
+        struct image_header *secondary_hdr = boot_img_hdr(state, slot);
+        uint32_t reset_value = 0;
+        uint32_t reset_addr = secondary_hdr->ih_hdr_size + sizeof(reset_value);
+
+        rc = flash_area_read(fap, reset_addr, &reset_value, sizeof(reset_value));
+        if (rc != 0) {
+            return BOOT_EFLASH;
```

Comment:
flash area should be closed before returning to the caller (by mcuboot internal policy).
Would by not i case if this bock will be converted to the function.

Reply:
I added a `goto` which should ensure that the area is closed.

I will try to come up with a helper function to see if that will makes things more clear.